apps/api_web/lib/api_web/event_stream.ex

```diff
@@ -26,43 +33,55 @@ defmodule ApiWeb.EventStream do
       |> put_resp_header("x-accel-buffering", "no")
       |> send_chunked(200)
 
-    {conn, pid}
+    ensure_timer(%__MODULE__{conn: conn, pid: pid, timeout: timeout})
   end
 
-  @spec event_stream_loop(loop, state) :: Plug.Conn.t() when loop: (loop, state -> Plug.Conn.t())
-  def event_stream_loop(next_call, {conn, pid} = state) do
-    with {:ok, new_conn} <- receive_result(state) do
-      next_call.(next_call, {new_conn, pid})
-    else
-      _ ->
-        Supervisor.server_unsubscribe(pid)
```

Comment:
It doesn't look like `Supervisor.server_unsubscribe/1` is ever called in the new code. Is it unnecessary with this new approach?

Reply:
Generally yes (the only way out terminates the process, which automatically unsubscribes), but it can't hurt to keep it in.